ludwig/backend/ray.py

```diff
@@ -160,6 +163,7 @@ def train_fn(
 
         train_shard = RayDatasetShard(
             rt.get_dataset_shard("train"),
+            train_dataset_size,
```

Comment:
This is the dataset size before being split among the workers. Have you tested this with multiple training workers?

I believe we need to also divide this by the number of workers, plus account for rounding (see Ray Datasets implementation of `split()`).

Reply:
@tgaddair Taking a look right now

Reply:
@tgaddair This would just be setting

```yaml
backend:
    trainer:
        num_workers: 2
```

or 

```yaml
backend:
    trainer:
        num_workers: 4
```

right? If yes, then this still works as intended with this fix.

I think the reason multiple workers isn't an issue is because we call the `ray.data.Dataset.split()` method on the DatasetPipeline objects, splitting the pipeline amongst multiple workers, rather than the dataset itself. So as long as we use the size of the dataset directly, we should be okay.

Would love it if someone can double check this understanding. @ShreyaR are you able to pull this branch and see if these changes fix the problem you were seeing before?

Reply:
I suppose what I don't understand is that the number of steps per epoch is dependent on the number of workers. So if you have 100 batches and 4 workers, there should be 25 steps per epoch. So how do we account for this in this approach?

In the previous implementation [here](https://github.com/ludwig-ai/ludwig/pull/2175/files#diff-db32a4692043fdf9957a872263f752a913c3cbb84a76ddaafa3c8dff6e2b5307L175), you can see that we're taking the length of the dataset pipeline after it has been split, which accounts for the number of workers.

Reply:
@tgaddair Gotcha, that makes sense. I'm able to reproduce what you said and can see that this implementation doesn't account for the number of workers and rounding. Will look into this again and create a fix that is in line with what you described.

Reply:
I wonder if it would be possible to window after splitting to work around this? Seems like it should be the same effect, more or less.